docs/hints.md

```diff
@@ -0,0 +1,45 @@
+# Hints
+
+Hints let you provide information on failing checks.
+
+![Hint Example](../misc/gh/ReadMe.png)
+
+Hints are a way to help make images more approachable.
+
+You can add a conditional hint or a check-wide hint. A conditional hint is printed when that specific condition is executed and fails. Make sure you understand the check precedence; this can be tricky, as sometimes your check is NOT executed ([read about conditions](conditions.md)).
+
+Example conditional hint:
+```
+[[check]]
+points = 5
+
+	[[check.pass]]
+	type = "ProgramInstalledNot"
+	name = "john"
+
+	[[check.pass]]
+	# This hint will NOT print unless the condition above succeeds.
+	# Pass conditions are logically AND-- they all need to succeed.
+	# If one fails, there's no reason to execute the other ones.
+	hint = "Removing just the binary is insufficient; use a package manager to remove all of a tool's files."
```

Comment:
at least by my logic, if a Program is NOT installed (so its no longer in the package manager DB saying it is or is not installed), then "using a package manager to remove the tool's files" would be useless since the package manager has no idea it exists -- if my understanding is right, shouldn't the hint be more like "The binary and its files need removed!"

Reply:
The example here is a "common" issue where people remove a package but don't purge or autoremove, so there's a hint for that (eg remove john, john-data still exists). I suppose the hint could be "Ensure that you use a package manager to remove ALL of a tool's files"